### PR TITLE
feat: create deployment wrap (MAPCO-3850)

### PR DESCRIPTION
**Pull request by ronenkapelian**
Dockerized + Helm charting

| Question                | Answer                                                                          |
| ---------------- | -------------------------------------------------------------------------- |
| Bug fix         | ✖                                                                        |
| New feature     | ✔                                                                        |
| Breaking change | ✖                                                                        |
| Deprecations    | ✖                                                                        |
| Documentation   | ✖                                                                        |
| Tests added     | ✖                                                                        |
| Chore            | ✔                                                                       |

This pull request will provide:
1. optimized Dockerfile:
  a: include mapproxy 1.16 installed
  b: node 20 env
  c: running on node:slim base image
2. helm chart including:
 a: secret creation
 b: tpl file based on raster-helm standard
3. refactor and remove unused openAPI sections inside the repo

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,32 +1,63 @@
...
+ENV PATH="$VIRTUAL_ENV/bin:$PATH"
+
+RUN mkdir -p /mapproxy && \
+    chown node:node /mapproxy
+RUN chmod -R 777 /mapproxy
+
 
 USER node
 EXPOSE 8080
 CMD ["dumb-init", "node", "--max_old_space_size=512", "./index.js"]
```
Comment (asafMasa) [inline Dockerfile:62]: why do you set the memory size to 512?

File: helm/templates/_resources.tpl [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+{{- define "configmap.fullname" -}}
+{{- printf "%s-%s-%s" .Release.Name .Chart.Name "configmap" | indent 1 }}
+{{- end }}
+
+{{/*
+Create deployment name as used by the service name label.
+*/}}
+{{- define "deployment.fullname" -}}
+{{- printf "%s-%s-%s" .Release.Name .Chart.Name "deployment" | indent 1 }}
+{{- end }}
```
Comment (asafMasa) [inline helm/templates/_resources.tpl:20]: end of file

File: helm/values.yaml [hunk 3]
```diff
@@ -53,15 +66,15 @@ caPath: '/usr/local/share/ca-certificates'
...
   tag: 'latest'
-  pullPolicy: IfNotPresent
+  pullPolicy: Always
 
 env:
   port: 8080
   targetPort: 8080
   protocol: TCP
-  logLevel: info
+  logLevel: debug
```
Comment (asafMasa) [inline helm/values.yaml:77]: should be info by default

File: helm/values.yaml [hunk 4]
```diff
@@ -71,38 +84,26 @@ env:
   metrics:
     enabled: false
     url: http://localhost:55681/v1/metrics
+  queue:
+    jobManagerBaseUrl: "http://localhost:8081"
+    heartbeat:
+      heartbeatManagerBaseUrl: "http://localhost:8089"
+      heartbeatIntervalMs: 3000
+    dequeueIntervalMs: 1000
```
Comment (asafMasa) [inline helm/values.yaml:92]: 2000

File: helm/values.yaml [hunk 5]
```diff
@@ -3,7 +3,20 @@ global:
   tracing: {}
   metrics: {}
   environment: ''
-
+  storage:
```
Comment (asafMasa) [inline helm/values.yaml:6]: is this section duplicated with line 11?

File: helm/values.yaml [hunk 6]
```diff
@@ -71,38 +84,26 @@ env:
...
+    heartbeat:
+      heartbeatManagerBaseUrl: "http://localhost:8089"
+      heartbeatIntervalMs: 3000
+    dequeueIntervalMs: 1000
+    jobType: "TilesSeeding"
+    tilesTaskType: "TilesSeeding"
+  mapproxy:
+    mapproxyApiUrl: "http://localhost:8083"
+  seedAttempts: 5
+  seedConcurrency: 5
```
Comment (asafMasa) [inline helm/values.yaml:98]: where is the storage type defintion is set? (FS or S3)

**Review by asafMasa — CHANGES_REQUESTED** [review]
Looking good, thanks!

File: .github/workflows/release-on-tag-push.yaml [hunk 7]
```diff
@@ -27,8 +27,6 @@ jobs:
           name: P_COLLECTION_FILE_NAME
           value: ${{ steps.package.outputs.name }}-v${{ steps.package.outputs.version }}-postman-collection.json
 
-      - run: 'npx openapi-to-postmanv2 -s openapi3.yaml -o ${{ env.P_COLLECTION_FILE_NAME }}'
-
```
[the remark this answers is not in the document]
Comment (ronenkapelian) [reply .github/workflows/release-on-tag-push.yaml:31]: done

File: helm/templates/_resources.tpl (hunk 2 above)
Comment (ronenkapelian) [reply helm/templates/_resources.tpl:20]: added

File: helm/values.yaml (hunk 5 above)
Comment (ronenkapelian) [reply helm/values.yaml:6]: as global mechanism, one is local level and one is global

File: helm/values.yaml (hunk 3 above)
Comment (ronenkapelian) [reply helm/values.yaml:77]: changed

File: helm/values.yaml (hunk 4 above)
Comment (ronenkapelian) [reply helm/values.yaml:92]: done

File: Dockerfile (hunk 1 above)
Comment (ronenkapelian) [reply Dockerfile:62]: as part of boilerplate

its affect on npm heap size acting with garbage-collection

https://stackoverflow.com/questions/48387040/how-do-i-determine-the-correct-max-old-space-size-for-node-js

seems to not be changed

File: helm/values.yaml (hunk 6 above)
Comment (ronenkapelian) [reply helm/values.yaml:98]: it's not controlled by the service, that storage type is related to the layer's source as defined on mapproxy.yaml,
so the cache-seeder only reads the tile source as provided on the configuration and executes seeding.
In theory its works, but if we would like it, so the deployment should support specific mount for tiles location (if the source is FS)

Comment (ronenkapelian) [reply helm/values.yaml:107]: after talking with you.
agreed and added FS full support

File: helm/templates/_tplValues.tpl [hunk 8]
```diff
@@ -52,3 +52,7 @@ Custom definitions
 {{- define "common.s3.merged" -}}
 {{- include "common.tplvalues.merge" ( dict "values" ( list .Values.storage.s3 .Values.global.storage.s3 ) "context" . ) }}
 {{- end -}}
+
+{{- define "common.fs.merged" -}}
+{{- include "common.tplvalues.merge" ( dict "values" ( list .Values.storage.fs .Values.global.storage.fs ) "context" . ) }}
+{{- end -}}
```
Comment (asafMasa) [inline helm/templates/_tplValues.tpl:58]: end of file

**Review by asafMasa — CHANGES_REQUESTED** [review]
looking good, please fix last minor comment

**Review by asafMasa — APPROVED** [review]

**Pull request merged**